packages/grid/x-data-grid/src/components/columnSelection/GridHeaderCheckbox.tsx

```diff
@@ -26,7 +26,7 @@ const useUtilityClasses = (ownerState: OwnerState) => {
   return composeClasses(slots, getDataGridUtilityClass, classes);
 };
 
-const GridHeaderCheckbox = React.forwardRef<HTMLInputElement, GridColumnHeaderParams>(
+const GridHeaderCheckbox = React.forwardRef<HTMLButtonElement, GridColumnHeaderParams>(
```

Comment:
@cherniavskii Would this consider a breaking change? I guess we don't have to change it in this PR for safety.

Reply:
Right, reverted now!